src/main/osd/osd_elements.c

```diff
@@ -321,16 +321,14 @@ static void osdFormatAltitudeString(char * buff, int32_t altitudeCm, osdElementT
     const char unitSymbol = osdGetMetersToSelectedUnitSymbol();
     unsigned decimalPlaces;
 
-    switch (variantType) {
-    case OSD_ELEMENT_TYPE_2:  // whole number altitude (no decimal places)
-        decimalPlaces = 0;
-        break;
-    case OSD_ELEMENT_TYPE_1:  // one decimal place (default)
-    default:
-        decimalPlaces = 1;
-        break;
-    }
-    osdPrintFloat(buff, SYM_ALTITUDE, osdGetMetersToSelectedUnit(altitudeCm) / 100.0f, "", decimalPlaces, true, unitSymbol);
+#ifdef USE_GPS
+    decimalPlaces = (variantType == OSD_ELEMENT_TYPE_2 || variantType == OSD_ELEMENT_TYPE_4) ? 0 : 1;
+    int32_t alt = (variantType == OSD_ELEMENT_TYPE_3 || variantType == OSD_ELEMENT_TYPE_4) ? getAltitudeAsl() : altitudeCm;
+#else
+    decimalPlaces = (variantType == OSD_ELEMENT_TYPE_2) ? 0 : 1;
```

Comment:
(decimals won't work if ASL is set in configurator, but FW is compiled without GPS support)

Reply:
Made a change in configurator for mitigation.

Reply:
Resolved in configurator